usecases/objects/auto_schema.go

```diff
@@ -32,26 +35,50 @@ import (
 	"github.com/weaviate/weaviate/usecases/objects/validation"
 )
 
-type autoSchemaManager struct {
+type AutoSchemaManager struct {
 	mutex         sync.RWMutex
 	schemaManager schemaManager
 	vectorRepo    VectorRepo
 	config        config.AutoSchema
 	logger        logrus.FieldLogger
+
+	// Metrics without labels to avoid cardinality issues
+	opsDuration  *prometheus.HistogramVec
+	tenantsCount prometheus.Counter
 }
 
-func newAutoSchemaManager(schemaManager schemaManager, vectorRepo VectorRepo,
+func NewAutoSchemaManager(schemaManager schemaManager, vectorRepo VectorRepo,
 	config *config.WeaviateConfig, logger logrus.FieldLogger,
-) *autoSchemaManager {
-	return &autoSchemaManager{
+	reg prometheus.Registerer,
+) *AutoSchemaManager {
+	r := promauto.With(reg)
+
+	tenantsCount := r.NewCounter(
+		prometheus.CounterOpts{
+			Name: "weaviate_auto_tenant_total",
+			Help: "Total number of tenants processed",
+		},
+	)
+
+	opDuration := r.NewHistogramVec(
```

Comment:
nit: shouldn't the variable name be `opsDuration`?

Reply:
we agreed to ignore it for now